File: db4on/core/src/core/com/db4o/MReleaseSemaphore.cs
namespace com.db4o
{
	internal sealed class MReleaseSemaphore : com.db4o.MsgD
	{
		internal sealed override bool ProcessMessageAtServer(com.db4o.foundation.network.YapSocket
			 sock)
		{
			string name = ReadString();
			((com.db4o.YapFile)GetStream()).ReleaseSemaphore(GetTransaction(), name);
			return true;
		}
	}
}

File: db4oj/core/src/com/db4o/MReleaseSemaphore.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o;

import com.db4o.foundation.network.YapSocket;

final class MReleaseSemaphore extends MsgD {
	public MReleaseSemaphore() {
		super();
	}

	public MReleaseSemaphore(MsgCloneMarker marker) {
		super(marker);
	}

	final boolean processMessageAtServer(YapSocket sock) {
		String name = readString();
		((YapFile)getStream()).releaseSemaphore(getTransaction(),name);
		return true;
	}
    
    public Object shallowClone() {
    	return super.shallowCloneInternal(new MReleaseSemaphore(MsgCloneMarker.INSTANCE));
    }
}